<?php
namespace App\Exceptions; use App\Library\Helper; use App\Library\Response; use Exception; use Illuminate\Foundation\Exceptions\Handler as ExceptionHandler; use Illuminate\Support\Facades\Log; class Handler extends ExceptionHandler { protected $dontReport = array(); protected $dontFlash = array('password', 'password_confirmation'); public function report(Exception $spb2d24a) { parent::report($spb2d24a); } private function msg($spa20801, $sp997a5c, $sp59ae99) { return $spa20801->isXmlHttpRequest() ? response()->json(array('message' => $sp997a5c), $sp59ae99) : response()->view('errors._', array('code' => $sp59ae99, 'message' => $sp997a5c), $sp59ae99); } public function render($spa20801, Exception $sp16b835) { if ($sp16b835 instanceof \Illuminate\Database\Eloquent\ModelNotFoundException) { return $this->msg($spa20801, trans('exception.resource_not_found'), 404); } elseif ($sp16b835 instanceof \Symfony\Component\HttpKernel\Exception\UnauthorizedHttpException) { Log::error('Caught a UnauthorizedHttpException', array('exception' => $sp16b835)); return $this->msg($spa20801, trans('exception.please_login'), 401); } elseif ($sp16b835 instanceof \Illuminate\Auth\Access\AuthorizationException) { return $this->msg($spa20801, trans('exception.un_authenticated'), 403); } elseif ($sp16b835 instanceof \Illuminate\Validation\ValidationException) { $spa20801->headers->set('X-Requested-With', 'XMLHttpRequest'); $spa20801->headers->set('Accept', 'application/json'); return parent::render($spa20801, $sp16b835); } elseif ($sp16b835 instanceof \Illuminate\Session\TokenMismatchException) { return $this->msg($spa20801, trans('exception.csrf_token_invalid'), 403); } elseif ($sp16b835 instanceof \Symfony\Component\HttpKernel\Exception\NotFoundHttpException) { return $this->msg($spa20801, trans('exception.not_found'), 404); } elseif ($sp16b835 instanceof \Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException) { return $this->msg($spa20801, trans('exception.method_not_allowed'), 405); } elseif ($sp16b835 instanceof \Symfony\Component\HttpKernel\Exception\ServiceUnavailableHttpException) { return $this->msg($spa20801, trans('exception.service_unavailable'), 503); } elseif ($sp16b835 instanceof \Symfony\Component\HttpKernel\Exception\HttpException) { $spe0111b = $sp16b835->getStatusCode() ?? 500; switch ($spe0111b) { case 429: return $this->msg($spa20801, trans('exception.too_many_requests'), $spe0111b); break; default: Log::error('Caught a UnknownHttpException', array('exception' => $sp16b835)); return $this->msg($spa20801, trans('exception.service_unavailable'), $spe0111b); } } elseif ($sp16b835 instanceof \Symfony\Component\HttpFoundation\Exception\SuspiciousOperationException) { return $this->msg($spa20801, trans('exception.method_not_allowed'), 405); } Log::error('Uncaught Exception', array('exception' => $sp16b835, 'method' => $spa20801->method(), 'url' => $spa20801->fullUrl(), 'data' => file_get_contents('php://input'), 'ip' => Helper::getIP() ?? $spa20801->ip(), 'client_ip' => $spa20801->getClientIp(), 'headers' => $spa20801->header())); if (config('app.debug')) { return parent::render($spa20801, $sp16b835); } else { return $this->msg($spa20801, trans('exception.unknown_error'), 500); } } }